packages/message-db/message-db/src/lib/Category.ts

```diff
@@ -112,70 +100,35 @@ export class MessageDbContext {
       snapshotStream,
       eventType,
     )
-    const decoded = await Snapshot.decode(decode, events)
+    const decoded = Snapshot.decode(decode, events)
     trace.getActiveSpan()?.setAttributes({
-      [Tags.snapshot_version]: decoded ? Number(decoded?.[0].version) : -1,
+      [Tags.snapshot_version]: decoded ? Number(decoded?.[0].version) : 0,
     })
     return decoded
   }
 
-  async reload<Event, State>(
```

Comment:
Reload is exactly the same as loadForwards, except it allows passing a position... I've made loadForwards accept the version so this method is obviated. This opens up some improvement opportunities in the `reload` method of the category which we'll implement in a later PR